Add participant groups feature

dodumosu opened the pull request:
this PR adds support for participant groups.

it also adds the `text/plain` mimetype for CSV imports.

dodumosu committed:
feat: initial support for participant groups

dodumosu committed:
fix: add the `text/plain` mimetype for CSV imports

dodumosu committed:
fix: add missing `name` column

dodumosu committed:
feat: filter submission lists by participant group

dodumosu committed:
feat: filter dashboard by participant group

dodumosu commented:
Added filtering submissions and the dashboard (when conflicts tracking is disabled in the latter case) by participant groups

dodumosu committed:
feat: add extra check

dodumosu committed:
fix: correct conditional expression

dodumosu committed:
feat: filter QA list using participant group

takinbo commented:
I've checked all the screens where this widget is supposed to be displayed and they all look adequately sized. There are only two nits that need to be resolved:

1. In all places where you've used "All Participant Groups" simply use "Participant Groups"
2. The styling of the widget needs to be consistent with the others. See screenshot below.

![D53D002F-E3D7-426E-BA3A-E9AEEB0463E1](https://user-images.githubusercontent.com/41004/177367097-2c93bd20-7c09-4448-9aa2-af1c49868bd2.jpeg)

dodumosu committed:
chore: change label and styling

dodumosu committed:
fix: update definition from source

dodumosu committed:
chore: do not initialize select2 on group widgets

dodumosu committed:
chore: use extended select field and add labels

this commit changes the empty label for location group filter fields
to a localized translation of "Location Group"
it also adds an empty label for participant group filter fields and
makes it single-choice, not multiple.

dodumosu committed:
chore: change rendering of participant groups
